frontend/src/app/plans/edit/[id]/_components/CreateNewPlanPage.tsx

```diff
@@ -80,11 +168,33 @@ export function CreateNewPlanPage({
     },
   });
 
+  const {
+    data: onlinePlan,
+    refetch: refetchOnlinePlan,
+    isLoading,
+  } = useQuery({
+    enabled: plan.onlineId !== null,
+    queryKey: ["onlinePlan", plan.onlineId],
+    queryFn: async () => {
+      const res = await getPlan({ id: Number(plan.onlineId) });
+      if (
+        res === false ||
+        (res as unknown as { status: number }).status === 404
+      ) {
+        plan.remove();
+        toast.error("Nie udało się pobrać planu");
+        router.push("/plans");
+        return null;
+      }
+      return res;
+    },
+  });
+
   const coursesFn = useMutation({
     mutationKey: ["courses"],
     mutationFn: async (registrationId: string) => {
       const response = await fetch(
-        `https://planer.solvro.pl/api/v1/departments/${faculty}/registrations/${encodeURIComponent(registrationId)}/courses`,
+        `${process.env.NEXT_PUBLIC_API_URL}/departments/${faculty}/registrations/${encodeURIComponent(registrationId)}/courses`,
```

Comment:
dlaczego nie używasz env.mjs

Reply:
bo dodales tam export usosowy ktory moze byc tylko po server side mistrzu, wiec albo bym to gdzies przeniosl i zmienial wszedzie albo tutaj po prostu uzylem zwyklego process.env